platform/http_uploader_apple.mm

```diff
@@ -64,23 +72,83 @@ - (void)uploadWithCompletion:(void (^)(NSInteger httpCode, NSString *description
 
   NSData * postData = [self requestDataWithBoundary:boundary];
 
-  NSURLSessionUploadTask * uploadTask = [[NSURLSession sharedSession]
-                                        uploadTaskWithRequest:uploadRequest
-                                        fromData:postData
-                                        completionHandler:^(NSData * data,
-                                                            NSURLResponse * response,
-                                                            NSError * error) {
+  NSURLSession * session =
+      [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]
+                                    delegate:self
+                               delegateQueue:nil];
+  NSURLSessionUploadTask * uploadTask = [session
+      uploadTaskWithRequest:uploadRequest
+                   fromData:postData
+          completionHandler:^(NSData * data, NSURLResponse * response, NSError * error) {
             NSHTTPURLResponse * httpResponse = (NSHTTPURLResponse *)response;
-            if (error == nil) {
+            if (error == nil)
+            {
               NSString * description = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
               completion(httpResponse.statusCode, description);
-            } else {
+            }
+            else
+            {
               completion(-1, error.localizedDescription);
             }
           }];
   [uploadTask resume];
 }
 
+- (void)URLSession:(NSURLSession *)session
+    didReceiveChallenge:(NSURLAuthenticationChallenge *)challenge
+      completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition disposition,
+                                  NSURLCredential * _Nullable credential))completionHandler
+{
+  NSString * authenticationMethod = challenge.protectionSpace.authenticationMethod;
+  if (authenticationMethod == NSURLAuthenticationMethodClientCertificate)
+  {
+    completionHandler(NSURLSessionAuthChallengeUseCredential, [self getClientUrlCredential]);
+  }
+  else if (authenticationMethod == NSURLAuthenticationMethodServerTrust)
+  {
+    NSURLCredential * credential =
```

Comment:
Put the implementation for "NSURLAuthenticationMethodServerTrust" under "#DEBUG" and use completionHandler(NSURLSessionAuthChallengePerformDefaultHandling, nil) for release implementation